PWGJE/EMCALJetTasks/UserTasks/AliAnalysisTaskJetExtractor.cxx

```diff
@@ -840,46 +874,11 @@ Bool_t AliAnalysisTaskJetExtractor::Run()
                                       truePtFraction,truePtFraction_PartLevel,fPtHard,fEventWeight,fImpactParameter);
     }
 
+    // perform the jet tagging
     if(fSaveTaggedInfo){
       Double_t taggedJetPt_Det = 0;
       Double_t taggedJetPt_Part = 0;
-      AliJetContainer * jetsHybrid = GetJetContainer(0);
-      AliJetContainer * jetsDetLevel = GetJetContainer(1);
-      AliJetContainer * jetsPartLevel = GetJetContainer(2);
-      // Validation
-      if (!jetsHybrid) {
-	AliErrorStream() << "Could not retrieve hybrid jet collection.\n";
-	return kFALSE;
-      }
-      if (!jetsDetLevel) {
-	AliErrorStream() << "Could not retrieve det level jet collection.\n";
-	return kFALSE;
-      }
-      if (!jetsPartLevel) {
-	AliErrorStream() << "Could not retrieve part level jet collection.\n";
-	return kFALSE;
-      }
-
-      // Now, begin the actual matching.
-      // Hybrid <-> det first
-      AliDebugStream(2) << "Matching hybrid to detector level jets.\n";
-      // First, we reset the tagging
-      ResetMatching(*jetsHybrid);
-      ResetMatching(*jetsDetLevel);
-      // Next, we perform the matching
-      PerformGeometricalJetMatching(*jetsHybrid, *jetsDetLevel, (jetsHybrid->GetJetRadius()*0.75));
-      // Now, begin the next matching stage
-      // det <-> particle
-      AliDebugStream(2) << "Matching detector level to particle level jets.\n";
-      // First, we reset the tagging. We need to reset the det matching again to ensure
-      // that it doesn't accidentally keep some latent matches to the hybrid jets.
-      ResetMatching(*jetsDetLevel);
-      ResetMatching(*jetsPartLevel);
-      // Next, we perform the matching
-      PerformGeometricalJetMatching(*jetsDetLevel, *jetsPartLevel, (jetsHybrid->GetJetRadius()*0.75));
-
-      GetJetTagging(*jetsHybrid, *jetsDetLevel, kFALSE, taggedJetPt_Det, taggedJetPt_Part);
-      GetJetTagging(*jetsDetLevel, *jetsPartLevel, kTRUE, taggedJetPt_Det, taggedJetPt_Part);
+      GetJetTagging(jet, *jetsHybrid, *jetsDetLevel, *jetsPartLevel, taggedJetPt_Det, taggedJetPt_Part);
```

Comment:
If you do put the matching behind a switch, you'll need to be careful to ensure that the jet collections are defined